index.js

```diff
@@ -110,8 +110,14 @@ class Conf {
 		}
 	}
 
+	_getkey(key) {
+		const escapeDot = value => value.replace(/\./g, '\\\\.');
+
+		return this._options.disableDotNotation ? escapeDot(key) : key;
+	}
+
 	get(key, defaultValue) {
-		return dotProp.get(this.store, key, defaultValue);
+		return dotProp.get(this.store, this._getkey(key), defaultValue);
```

Comment:
Instead of escaping the dot, simply don't use `dotProp` when dot-notation is turned off.